### PR TITLE
Adds the ability to set an associated identifier for use with Connect.

### PR DESCRIPTION
**Pull request by gatlinhebert**
Does not include the helper methods for setting advertising ids and properties.

### CODE REVIEW

**Comment by rlepinski** [comment]
👍  Nice

File: src/android/UAirshipPlugin.java [hunk 1]
```diff
@@ -756,6 +756,28 @@ void isAnalyticsEnabled(JSONArray data, CallbackContext callbackContext) {
     }
 
     /**
+     * Sets associated custom identifiers for use with the Connect data stream.
+     * <p/>
+     * Previous identifiers will be replaced by the new identifiers each time associateIdentifiers is called. It is a set operation.
```
Comment (hcrowell) [inline src/android/UAirshipPlugin.java]: `associateIdentifiers` -> `setAssociatedIdentifier` ?

File: src/ios/UAirshipPlugin.h [hunk 2]
```diff
@@ -224,6 +224,17 @@
 - (void)setAnalyticsEnabled:(CDVInvokedUrlCommand *)command;
 
 /**
+ * Sets associated custom identifiers for use with the Connect data stream.
+ *
+ * Previous identifiers will be replaced by the new identifiers each time associateIdentifiers is called. It is a set operation.
```
Comment (hcrowell) [inline src/ios/UAirshipPlugin.h]: `associateIdentifiers` -> `setAssociatedIdentifier` ?

File: www/UrbanAirship.js [hunk 3]
```diff
@@ -444,6 +444,20 @@ module.exports = {
     return new TagGroupEditor('editChannelTagGroups')
   },
 
+    /**
```
Comment (hcrowell) [inline www/UrbanAirship.js]: align this one space left*

File: www/UrbanAirship.js [hunk 4]
```diff
@@ -444,6 +444,20 @@ module.exports = {
...
+   * Sets an associated identifer for the Connect data stream.
```
Comment (hcrowell) [inline www/UrbanAirship.js]: identifier*

File: www/UrbanAirship.js [hunk 5]
```diff
@@ -444,6 +444,20 @@ module.exports = {
...
+   *
+   * @param {string} Custom key for identifier.
+   * @param {string} The identifer value.
```
Comment (hcrowell) [inline www/UrbanAirship.js]: identifier*

File: www/UrbanAirship.js [hunk 6]
```diff
@@ -444,6 +444,20 @@ module.exports = {
...
+   * @param {function} [success] Success callback.
+   * @param {function(message)} [failure] Failure callback.
+   * @param {string} failure.message The error message.
+   */
+  setAssociatedIdentifier: function(key, identifer, success, failure) {
```
Comment (hcrowell) [inline www/UrbanAirship.js]: identifier*

File: www/UrbanAirship.js [hunk 7]
```diff
@@ -444,6 +444,20 @@ module.exports = {
...
+    argscheck.checkArgs('ssFF', 'UAirship.setAssociatedIdentifier', arguments)
+    callNative(success, failure, "setAssociatedIdentifier", [key, identifer])
```
Comment (hcrowell) [inline www/UrbanAirship.js]: identifier*

**Comment by hcrowell** [comment]
🍊  after fixing typos above ^. Nice job! :)

**Pull request merged**